packages/plugins/i18n/server/services/entity-service-decorator.js

```diff
@@ -169,16 +169,25 @@ const decorator = (service) => ({
 
     const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
 
-    const query = transformParamsToQuery(uid, wrappedParams);
-
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
-        return strapi.db.query(uid).findMany(query);
+        return service.findMany.call(this, uid, wrappedParams);
+      }
+      const query = {
+        ...transformParamsToQuery(uid, wrappedParams),
+        select: [],
+        populate: {},
+      };
+
+      const result = await strapi.db.query(uid).findOne(query);
+      if (result === null) {
+        return null;
       }
-      return strapi.db.query(uid).findOne(query);
+      // Since we change from findMany to findOne we need to restart the process so we use the entityService
+      return strapi.entityService.findOne(uid, result.id);
     }
 
-    return strapi.db.query(uid).findMany(query);
+    return service.findMany.call(this, uid, wrappedParams);
```

Comment:
the entityService find many is already calling wrappedParams, so I think it's not necessary to wrap it twice